publish-docs.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+set -e
+
+SRCDIR=$(dirname "$0")
+BUILDDIR="$SRCDIR/build"
+DRY_RUN=false
+
+ARGS=$(getopt -n "$0" -o d:nh -l builddir:,dry-run,help -- "$@")
+eval set -- "$ARGS"
+
+usage() {
+    cat <<EOF
+Usage: $0 [OPTION...] BUCKET REGION
+
+Publish HTML to documentation to S3 bucket
+
+  -d, --builddir	path to HTML build directory (default: $BUILDDIR)
+  -n, --dry-run		only show what would be published
+  -h, --help		show this help message an exit
+EOF
+}
+
+while true; do
+    case "$1" in
+	-d|--builddir)
+	    BUILDDIR=$2
+	    shift 2
+	    ;;
+        -n|--dry-run)
+            DRY_RUN=true
+            shift
+            ;;
+        -h|--help)
+            usage
+            exit 0
+            ;;
+        --)
+            shift
+            break
+            ;;
+        *)
+            echo "error: Unrecognized option \"$1\"" >&2
+            exit 1
+            ;;
+    esac
+done
+
+if [ $# -lt 2 ]; then
+    echo "error: BUCKET and REGION required" >&2
```

Comment:
I think that region should be an option. It can be set in the AWS CLI profile or in an environment variable (like the credentials).

Reply:
That's fair.